SharpPcap/DeviceConfiguration.cs

```diff
@@ -76,16 +78,30 @@ internal void RaiseConfigurationFailed(string property, Exception exception)
                 Property = property,
                 Error = PcapError.Generic,
                 Message = message,
+                Ex = exception,
             };
-            ConfigurationFailed?.Invoke(this, args);
+
+            RaiseFailedEvent(args);
         }
 
+        private void RaiseFailedEvent(ConfigurationFailedEventArgs args)
+        {
+            if (ConfigurationFailed is null)
+            {
+                throw args.Ex;
+            }
+            else
+            {
+                ConfigurationFailed.Invoke(this, args);
+            }
+        }
     }
 
     public class ConfigurationFailedEventArgs : EventArgs
     {
         public PcapError Error { get; internal set; }
         public string Property { get; internal set; }
         public string Message { get; internal set; }
+        public Exception Ex { get; internal set; }
```

Comment:
```suggestion
        public Exception Error { get; internal set; }
```
See also https://docs.microsoft.com/en-us/dotnet/api/system.io.erroreventargs?view=net-5.0

Reply:
Wouldn't that overlap with the existing PcapError Error parameter? Should we rename PcapError Error to PcapError PcapError?

Reply:
Then rename it to `Exception`
```suggestion
        public Exception Exception { get; internal set; }
```